File: python/n342_PowerOfFour.py
# https://leetcode-cn.com/problems/power-of-four/
class Solution:
    def isPowerOfThree(self, num: int) -> bool:
        if num <= 0: return False
        while num > 1 and num % 4 == 0:
            num = num / 4
        return num == 1


if __name__ == "__main__":
    solu = Solution()
    print(solu.isPowerOfThree(28))

File: cpp/n342_PowerOfFour.cpp
// https://leetcode-cn.com/problems/power-of-four/

#include <iostream>
#include <vector>
#include <sstream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool isPowerOfThree(int num) {
        if (num <= 0) return false;
        while (num > 1 && num % 4 == 0) num = num / 4;
        return num == 1;
    }
};

int main(){
    Solution solu;
    cout << solu.isPowerOfThree(28) << endl;
}